### DISCUSSION TITLE
Nextjs Group Routing in Production

### DISCUSSION THREAD

**Discussion by PatrickNaenae** [vercel/next.js#51701] [Help] [answered]
### Summary

When i deploy my nextjs app, the site breaks when i try to navigate through the links and i noticed it's because i grouped my routes. When i removed the grouping, it worked. I added it back and the error came back. The issue with removing the group routing is that it adds the group folder to the url which i don't want. For example, I want the url to be mozfin.com/about not mozfin.com/main/about where (main) is the folder i want to be the group folder. 
![Screen Shot 2023-06-23 at 10 32 58](https://github.com/vercel/next.js/assets/83951220/88884a38-c950-4a80-a7f7-de14e8a9ec48)
![Screen Shot 2023-06-23 at 10 34 20](https://github.com/vercel/next.js/assets/83951220/07126511-d150-49da-b17f-3d343bc77a4c)

I want the main and registration to be group routing. Thank you

**Comment by devjiwonchoi** [comment]
The Route Groups need to be in the format of `(group)` with parentheses around.

In your case, try to name the folder as `(main)`.

## Reference
- [Route Group Convention](https://nextjs.org/docs/app/building-your-application/routing/route-groups#convention)

**Reply by PatrickNaenae** [reply]
Yeah. That was how i did it initially. It worked locally but when i moved it to production, the links break. I had to remove the parentheses and it worked again but this time, it added the 'main' directory to the url 'mozfin.com/main/about. I know that if i group it, the 'main' wont be added to the url, but doing it like that breaks the code in production

**Reply by devjiwonchoi** [reply]
Do you have a Dynamic Route on your root route? Seems like the root route is catching all `/:path`.

**Reply by PatrickNaenae** [reply]
No I don't. They are all static routes

**Reply by devjiwonchoi** [reply]
Could you share your overall folder structure?

**Comment by PatrickNaenae** [comment]
![Screen Shot 2023-06-23 at 11 58 47](https://github.com/vercel/next.js/assets/83951220/291cb610-6063-459d-b96c-0a4a6389b3c4)
![Screen Shot 2023-06-23 at 11 59 01](https://github.com/vercel/next.js/assets/83951220/fe9d3a5c-6819-4e57-854b-c9dc6d469ab7)

That is it

**Reply by devjiwonchoi** [reply]
That is weird. Have you configured anything that rewrites the main page to `/:path`, or is it unexpected behavior? Such as [mozfin.com/test](https://mozfin.com/test).

**Reply by PatrickNaenae** [reply, marked as answer]
No. I didn't configure the main page. But the help desk of the hosting platform configured the web.config file and that is what is redirecting every get request to the homepage url. 
This is what is in the web.config file
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.web>
       
    </system.web>
    <system.webServer>
        <rewrite>
        <rules>

            <rule name="index" patternSyntax="Wildcard">
                <match url="*" />
                    <conditions>
                        <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
                        <add input="{REQUEST_FILENAME}" matchType="IsDirectory" negate="true" />
                    </conditions>
                <action type="Rewrite" url="index.html" />
            </rule>

        </rules>
    </rewrite>

    </system.webServer>
</configuration>
